MoreLinq.Test/DuplicatesTest.cs

```diff
@@ -0,0 +1,220 @@
+        {
+            var stringArray = new[]
+            {
+                "FirstElement",
+                "SecondElement",
+                "ThirdElement"
+            };
+
+            var duplicates = stringArray.Duplicates();
+
+            Assert.That(duplicates, Is.Empty);
+        }
+
+        [Test]
+        public void When_Asking_For_Duplicates_On_Sequence_Projection_Without_Duplicates_Then_Then_Empty_Sequence_Is_Returned()
+        {
+            var dummyClasses = new DummyClass[]
+            {
+                new("FirstElement"),
+                new("SecondElement"),
+                new("ThirdElement")
+            };
+
+            var duplicates = dummyClasses.Duplicates(x => x.ComparableString);
+
+            Assert.That(duplicates, Is.Empty);
+        }
+
+        [Test]
+        public void When_Asking_For_Duplicates_On_Sequence_With_Duplicates_Then_True_Is_Returned()
+        {
+            var stringArray = new[]
+            {
+                "FirstElement",
+                "DUPLICATED_STRING",
+                "DUPLICATED_STRING",
+                "ThirdElement"
+            };
+
+            var duplicates = stringArray.Duplicates();
+
+            Assert.That(duplicates, Contains.Item("DUPLICATED_STRING"));
+        }
+
+        [Test]
+        public void When_Asking_For_Duplicates_On_Sequence_Projection_With_Duplicates_Then_True_Is_Returned()
+        {
+            var dummyClass = new DummyClass("DUPLICATED_STRING");
+            var dummyClasses = new DummyClass[]
+            {
+                new("FirstElement"),
+                dummyClass,
+                dummyClass,
+                new("ThirdElement")
+            };
+
+            var duplicates = dummyClasses.Duplicates(x => x.ComparableString);
+
+            Assert.That(duplicates, Contains.Item(dummyClass));
+        }
+
+        [Test]
+        public void When_Asking_For_Duplicates_On_Sequence_With_Duplicates_Then_It_Does_Not_Iterate_Unnecessary_On_Elements()
+        {
+            var source = MoreEnumerable.From(() => "FirstElement",
+                () => "DUPLICATED_STRING",
+                () => "DUPLICATED_STRING",
+                () => throw new TestException());
+
+            Assert.DoesNotThrow(() => source.Duplicates());
+        }
+
+        [Test]
+        public void When_Asking_For_Duplicates_On_Sequence_Projection_With_Duplicates_Then_It_Does_Not_Iterate_Unnecessary_On_Elements()
+        {
+            var source = MoreEnumerable.From(() => new DummyClass("FirstElement"),
+                () => new DummyClass("DUPLICATED_STRING"),
+                () => new DummyClass("DUPLICATED_STRING"),
+                () => throw new TestException());
+
+            Assert.DoesNotThrow(() => source.Duplicates(x => x.ComparableString));
+        }
+
+        [Test]
+        public void When_Asking_Duplicates_Then_It_Is_Executed_Right_Away()
```

Comment:
Is it executed right away? It's only because you're call `Consume` and forcing its execution so I don't think this test is demonstrating any unique behaviour. What it should be testing, instead, is that is has _deferred execution semantics_ (i.e. lazily evaluated).

Reply:
done